### ISSUE TITLE
Provide option for ignoring missing directories/files

### ISSUE THREAD

**Issue by stefan6419846** [PyCQA/flake8#1656]
### describe the request

I recently started to employ *flake8* for linting the code of some large legacy code base. As there are tons of code to check and fix, this will take some time, so some modules will already be fixed, while others are not.

Now I want to make sure that already fixed modules stay like this by employing partial checks in the corresponding CI environment. The approach would be to just pass the fixed module paths to *flake8* explicitly as this is the easiest approach.

As there are different projects which do not use all of the modules, this becomes an issue: As soon as I pass an invalid path to *flake8*, the return code will be 1 instead of 0, while the actual linting has been successful. To circumvent this, I searched for a corresponding CLI option, but could not find a suitable one. I probably could write my own wrapper for this to handle such cases, but I assume that this might be of general interest.

For this reason, I would like to see an option for getting an exit code 0 iff the actual linting did not find any issues to report, ignoring possible missing directories/files.

**Comment by asottile** [comment]
> As there are different projects which do not use all of the modules

I don't think I understand this?

passing something that doesn't exist should be an error -- I don't think we should change this

it would be trivial for you to write a tiny wrapper script that filters out files that don't exist

or utilize a blocklist instead of an allowlist would also work

**Comment by asottile** [comment]
> This probably is a matter of taste - and given a dedicated option this would not affect the default behavior

yeah we can't just say "yes" to every option -- you end up with a ratsnest of unmaintainable "features" that nobody uses.

> Yes, writing a wrapper should definitely not be this hard. Using an allow list as an alternative should be much easier than a block list, as I rather know which packages have been migrated instead of keeping track of all the packages I have yet missed.

I am speaking from experience -- you actually want a blocklist and not an allowlist so that new files adhere to the rules rather than your problem silently getting worse over time

**Comment by sigmavirus24** [comment]
Also, don't we still have `--exit-zero`?

**Comment by stefan6419846** [comment]
According to the documentation, `--exit-zero` will always return 0, even if coding style violations have occurred, thus it cannot really be used in this case.

**Comment by sigmavirus24** [comment]
Verified by scanning the code quickly from my phone. Just use that option and read the help more closely next time

**Comment by sigmavirus24** [comment]
I fat-fingered the lock button on my phone. Unlocking because it didn't need to be locked. Thanks for pinging @stefan6419846

**Comment by stefan6419846** [comment]
Thanks for unlocking.

As already mentioned inside my last comment here, `--exit-zero` is not suitable in this case. According to the [docs](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-exit-zero), this option will return exit code 0 even if there are style violations (which is backed by looking at the source code as well, where only "catastrophic failures" lead to exit code 1): https://github.com/PyCQA/flake8/blob/e249dc47dfd314c34bbd0946c3fad970a8b15d6c/src/flake8/main/application.py#L99-L107

This does not suit my use case, as I only wanted to suppress the *FileNotFoundError*. In the meantime, I saw that disabling *pycodestyle* error code *E902* using `flake8 --ignore=E902 missing_directory/` might fix this, although this would catch all general "IOError"s, which might not be suitable.

Nevertheless, I already settled with a custom wrapper which carries a corresponding up-to-date list, as mentioned/proposed earlier.
